src/cantools/database/can/message.py

```diff
@@ -750,36 +750,32 @@ def _assert_signal_values_valid(self,
                 if signal_value_num is None:
                     raise EncodeError(f'Invalid value specified for signal '
                                       f'"{signal.name}": "{signal_value}"')
-
                 continue
 
-            if signal.minimum is not None:
-                min_effective = signal.minimum
+            if scaling:
+                scaled_value = signal_value
+            else:
+                if signal.conversion.choices and signal_value in signal.conversion.choices:
```

Comment:
that condition probably must go one level up (scaled values can also be enums)

Reply:
That's true, fixed.